.github/actions/fetch-nightly-cli/action.yaml

```
@@ -0,0 +1,58 @@
+name: Fetch nightly CLI
+description: From an outside repo, fetch the latest nightly CLI build from Wasp's CI runs.
+
+inputs:
+  branch:
+    description: The branch to fetch the nightly build from. Defaults to 'main'.
+    required: false
+    default: main
+  token:
+    description: GitHub token to authenticate the request.
+    required: false
+    default: ${{ github.token }}
+  output-dir:
+    description: Directory to download the CLI build into. Defaults to the current directory.
+    required: false
+    default: "."
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Get latest successful run ID
+      id: get_run_id
+      shell: bash
+      run: |
+        run_id=$(
+          gh run list \
+            --workflow "$workflow_name" \
+            --branch "$branch" \
+            --status success \
+            --limit 1 \
+            --json databaseId \
+            --jq '.[0].databaseId'
+        )
+        echo "run_id=$run_id" >> $GITHUB_OUTPUT
+      env:
+        GH_REPO: wasp-lang/wasp
+        GH_TOKEN: ${{ inputs.token }}
+        branch: ${{ inputs.branch }}
+        workflow_name:
+          # This name must be in sync with the name used in:
+          # /.github/workflows/ci.yaml
+          "CI"
+
+    - name: Download artifact
+      if: matrix.args.needs-download == true
+      shell: bash
+      run: |
+        gh run download "$run_id" --name "$artifact_name" --dir "$output_dir"
+      env:
+        GH_REPO: wasp-lang/wasp
+        GH_TOKEN: ${{ inputs.token }}
+        run_id: ${{ steps.get_run_id.outputs.run_id }}
+        output_dir: ${{ inputs.output-dir }}
+        artifact_name:
```

I'd also go with all uppercase names:

```suggestion
        RUN_ID: ${{ steps.get_run_id.outputs.run_id }}
        OUTPUT_DIR: ${{ inputs.output-dir }}
        ARTIFACT_NAME:
```

---

I tried to match the energy of the shell, where uppercase names are globals or env vars for other programs, and lowercase names are the script's own variables.